docs/metasploit-framework.wiki/Metasploit-Filesystem-Explained.md

```diff
@@ -0,0 +1,79 @@
+with the file system of Metasploit for most use cases.
+
+## app
+This directory contains ActiveDirectory concerns, models and validators.
+
+## config
+Contains various files that help configure Metasploit. Most files here you'll never have to deal with, though 
+`database.yml.example` might be useful for those looking to configure their database, and `openssl.conf` 
+might be helpful for those trying to troubleshoot OpenSSL issues in Metasploit.
+
+## data
+This folder contains various data files used for a variety of purposes, including but not limited to banners for the
+console, exploit source code for exploits (under `data/exploits`), template code and binaries, wordlists and shellcode.
+
+As a general rule of thumb this folder will most often be used when you are using compiled binaries or source code from
+other exploits for cases such as local privilege escalation exploits and need to provide the exploit code and compiled 
+binaries so that maintainers can verify the binary and compile it themselves, as so that modules can find the R7 compiled
+version of the resulting binary for use during exploitation.
+
+## db
+Contains `modules_metadata_base.json` which contains information about all modules within Metasploit, as well as 
+`schema.rb` which describes some of the fields within the Metasploit database.
+
+## docker
+Contains files related to running Metasploit inside Docker.
+
+## docs
+This contain all the documentation files that are on https://docs.metasploit.com as well as the associated server files
+so that you can run a local copy of this site offline.
+
+## documentation
+This folder is primarily used to hold documentation for Metasploit's various modules, as well as the developers guide
+at `developers_guide.pdf`.
+
+## external
+This folder contains files to assist in using Metasploit with other tools such as `burp-proxy`, `vscode` and `zsh`.
+
+## kubernetes
+Contains files related to deploying Metasploit in Kubernetes for Kubernetes testing, using a `meterpreter` helm chart.
+
+## lib
+This is where all of the libraries and mixins of Metasploit live. Generally speaking any code that will be reused
+across multiple modules will be placed into a library or mixin which will then be placed under this folder.
+
+## modules
+This is the folder where all of Metasploit's modules live. These modules are scripts in Ruby that interface with
+Metasploit itself to perform some specific task. There are various types of modules, such as `exploit` modules to
+exploit a vulnerability and gain a shell, `auxiliary` to perform a non-shell gaining activity, `payloads` for
+Metasploit's various payloads (which are also modules), and `post` for post exploitation modules.
+
+## plugins
+This is where plugins for Metasploit live, which allow Metasploit to interface with other tools and services.
+
+## script
+Old folder, will likely get removed in the future.
```

Comment:
This folder looks abandoned; its possible we should just remove it and then remove it from our documentation.